ujit_codegen.c

```diff
@@ -159,6 +159,36 @@ ujit_side_exit(jitstate_t* jit, ctx_t* ctx)
     return code_ptr;
 }
 
+#if RUBY_DEBUG
+
+#define GEN_COUNTER_INC(cb, counter_name) _gen_counter_inc(cb, &(ujit_runtime_counters . counter_name))
+static void
+_gen_counter_inc(codeblock_t *cb, int64_t *counter)
+{
+    if (!rb_ujit_opts.gen_stats) return;
+     mov(cb, REG0, const_ptr_opnd(counter));
+     // XXX: Will want lock prefix for ractors.
+     add(cb, mem_opnd(64, REG0, 0), imm_opnd(1));
```

Comment:
I've never done this but there might be a way to do this with an atomic operation, no lock required?

Reply:
Nice!